include/deal.II/lac/trilinos_vector.h

```diff
@@ -988,7 +988,7 @@ namespace internal
                                 TrilinosWrappers::MPI::Vector &v,
                                 bool fast)
       {
-        v.reinit(matrix.range_partitioner(), fast);
+        v.reinit(IndexSet(matrix.range_partitioner()), MPI_COMM_WORLD, fast);
```

Comment:
COMM_WORLD is probably not always correct. Sadly SparseMatrix has no way to query the communicator. Can you do something like ```matrix.range_partitioner().Comm().Comm()``` (maybe need to cast to Epetra_MpiComm). Or introduce a way to query the SparseMatrix for the communicator (PETSc has ```get_mpi_communicator()```)